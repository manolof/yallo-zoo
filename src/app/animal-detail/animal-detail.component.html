<div class="my-8 px-8">
	<ng-container *ngIf="animal$ | async as animal; else loading">
		<figure class="bg-white shadow-md rounded-lg transition-shadow">
			<div class="relative pb-40">
				<img
					class="absolute inset-0 h-full w-full object-cover rounded-t-lg"
					[src]="animal.avatar"
					[alt]="animal.name"
				/>
			</div>
			<figcaption class="p-4">
				<h1 class="mt-2 mb-2 text-2xl font-bold">
					{{ animal.name }}
				</h1>
				<p class="text-lg">
					{{ animal.type }}
				</p>
			</figcaption>
		</figure>
	</ng-container>
	<ng-template #loading>
		<div class="px-8">Loading...</div>
	</ng-template>
</div>
