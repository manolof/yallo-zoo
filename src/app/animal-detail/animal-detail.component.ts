import {
	ChangeDetectionStrategy,
	Component,
	inject,
	OnInit,
} from '@angular/core';
import { ActivatedRoute, ParamMap } from '@angular/router';
import { Observable } from 'rxjs';

import { HttpService } from '../http.service';
import { Animal } from '../animal';

@Component({
	selector: 'app-animal-detail',
	templateUrl: './animal-detail.component.html',
	changeDetection: ChangeDetectionStrategy.OnPush,
})
export class AnimalDetailComponent implements OnInit {
	public animal$!: Observable<Animal>;

	private route = inject(ActivatedRoute);
	private httpService = inject(HttpService);

	public ngOnInit() {
		this.route.paramMap.subscribe((params: ParamMap) => {
			const id = params.get('id')!;
			this.animal$ = this.httpService.getAnimalDetail(id);
		});
	}
}
