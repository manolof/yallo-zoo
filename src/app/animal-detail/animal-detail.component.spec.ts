import { ComponentFixture, TestBed } from '@angular/core/testing';
import { ActivatedRoute, convertToParamMap } from '@angular/router';
import { of } from 'rxjs';

import { AnimalDetailComponent } from './animal-detail.component';
import { HttpService } from '../http.service';

describe('AnimalDetailComponent', () => {
	let component: AnimalDetailComponent;
	let fixture: ComponentFixture<AnimalDetailComponent>;

	const httpServiceMock = {
		getAnimalDetail: jest
			.fn()
			.mockReturnValue(
				of({ id: 1, name: 'Lion', type: 'Mammal', avatar: 'lion.jpg' }),
			),
	};

	beforeEach(async () => {
		await TestBed.configureTestingModule({
			declarations: [AnimalDetailComponent],
			providers: [
				{
					provide: ActivatedRoute,
					useValue: { paramMap: of(convertToParamMap({ id: 1 })) },
				},
				{ provide: HttpService, useValue: httpServiceMock },
			],
		}).compileComponents();

		fixture = TestBed.createComponent(AnimalDetailComponent);
		component = fixture.componentInstance;
	});

	it('should create', () => {
		expect(component).toBeTruthy();
	});

	it('should get animal detail from http service', () => {
		component.ngOnInit();
		fixture.detectChanges();
		expect(fixture).toMatchSnapshot();
	});
});
