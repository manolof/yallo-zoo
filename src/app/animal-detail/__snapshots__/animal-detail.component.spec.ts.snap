// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AnimalDetailComponent should get animal detail from http service 1`] = `
<app-animal-detail
  __ngContext__={[Function Number]}
  animal$={[Function Observable]}
  httpService={[Function Object]}
  route={[Function Object]}
>
  <div
    class="my-8 px-8"
  >
    <figure
      class="bg-white shadow-md rounded-lg transition-shadow"
    >
      <div
        class="relative pb-40"
      >
        <img
          alt="Lion"
          class="absolute inset-0 h-full w-full object-cover rounded-t-lg"
          src="lion.jpg"
        />
      </div>
      <figcaption
        class="p-4"
      >
        <h1
          class="mt-2 mb-2 text-2xl font-bold"
        >
           Lion 
        </h1>
        <p
          class="text-lg"
        >
           Mammal 
        </p>
      </figcaption>
    </figure>
  </div>
</app-animal-detail>
`;
